File: frontend/src/types/risk.ts
// Risk types
export interface RiskFlag {
  id: string;
  document_id: string;
  risk_level: 'critical' | 'high' | 'medium' | 'low' | 'info';
  category: string;
  description: string;
  recommendation: string;
  evidence_ref: string | null;
}

export interface RiskOverview {
  total_flags: number;
  critical: number;
  high: number;
  medium: number;
  low: number;
  risk_score: number;
  flags: RiskFlag[];
}

File: backend/app/schemas/risk.py
"""Risk schemas."""

from pydantic import BaseModel
from typing import List, Optional


class RiskFlag(BaseModel):
    id: str
    document_id: str
    risk_level: str  # critical | high | medium | low | info
    category: str
    description: str
    recommendation: str
    evidence_ref: Optional[str] = None


class RiskOverview(BaseModel):
    total_flags: int
    critical: int
    high: int
    medium: int
    low: int
    risk_score: float
    flags: List[RiskFlag]
